toponetx/algorithms/components.py

```diff
@@ -0,0 +1,220 @@
+"""Module to compute connected components on topological domains."""
+from collections.abc import Collection, Hashable, Iterable, Iterator
+from warnings import warn
+
+import networkx as nx
+import numpy as np
+
+from toponetx.classes.cell import Cell
+from toponetx.classes.cell_complex import CellComplex
+from toponetx.classes.colored_hypergraph import ColoredHyperGraph
+from toponetx.classes.combinatorial_complex import CombinatorialComplex
+from toponetx.classes.simplicial_complex import SimplicialComplex
+from toponetx.exception import TopoNetXError
+
+__all__ = [
+    "s_connected_components",
+    "s_component_subcomplexes",
+    "connected_components",
+    "connected_component_subcomplexes",
+]
+
+
+def s_connected_components(
+    complex, s: int = 1, cells: bool = True, return_singletons: bool = False
+):
+    """Return generator for the s-connected components.
+
+    Parameters
+    ----------
+    complex : Complex
+        Supported complexes are cell/combintorial and hypegraphs.
+    s : int, optional
+        The number of intersections between pairwise consecutive cells.
+    cells : bool, optional
+        If True will return cell components, if False will return node components.
+    return_singletons : bool, optional
+
+    Notes
+    -----
+    If cells=True, this method returns the s-cell-connected components as
+    lists of lists of cell uids.
+    An s-cell-component has the property that for any two cells e1 and e2
+    there is a sequence of cells starting with e1 and ending with e2
+    such that pairwise adjacent cells in the sequence intersect in at least
+    s nodes. If s=1 these are the path components of the cell complex.
+
+    If cells=False this method returns s-node-connected components.
+    A list of sets of uids of the nodes which are s-walk connected.
+    Two nodes v1 and v2 are s-walk-connected if there is a
+    sequence of nodes starting with v1 and ending with v2 such that pairwise
+    adjacent nodes in the sequence share s cells. If s=1 these are the
+    path components of the cell complex .
+
+    Yields
+    ------
+    s_connected_components : iterator
+        Iterator returns sets of uids of the cells (or nodes) in the s-cells(node)
+        components of the cell complex.
+
+    Example
```

Comment:
NumPy's documentation style dictates that this headline should read `Examples` even if there is only one (same throughout this file).

Reply:
```
Examples
----------
```